features/player_stats.lua

```diff
@@ -166,12 +169,28 @@ local function player_console_chat(event)
     end
 end
 
-local function player_built_entity()
-    change_for_global(built_by_players_name, 1)
+local function player_built_entity(event)
+    local player_index = event.player_index
+    if event.created_entity.is_registered_for_construction() == false then -- When is_registered_for_construction() is true we only register the entity as built once a robot builds it
+        change_for_global(built_by_players_name, 1)
+        change_for_player(player_index, player_entities_built_name, 1)
+    end
 end
 
-local function robot_built_entity()
+local function robot_built_entity(event)
     change_for_global(built_by_robots_name, 1)
+    local robot = event.robot
+
+    if not robot then
```

Comment:
```suggestion
    if not robot.valid then
```
I was worried about another event handler destroying the robot, not the field being removed from the event table.